tests/mbt/driver/mbt_test.go

```diff
@@ -0,0 +1,577 @@
+		case "DeliverVscMaturedPacket":
+			consumerChain := lastAction["consumerChain"].Value.(string)
+			t.Log("DeliverVscMaturedPacket", consumerChain)
+
+			var expectError bool
+			if ConsumerStatus(currentModelState, consumerChain) == "timedout" {
+				expectError = true
+				// expectError is true, because we expect the consumer to have timed out.
+				// the call will fail if there is no error
+				driver.DeliverPacketFromConsumer(ChainId(consumerChain), expectError)
+
+				// stop the consumer chain on the provider
+				driver.providerKeeper().StopConsumerChain(driver.providerCtx(), consumerChain, expectError)
+			} else {
+				expectError = false
+				driver.DeliverPacketFromConsumer(ChainId(consumerChain), expectError)
+			}
+		default:
+
+			log.Fatalf("Error loading trace file %s, step %v: do not know action type %s",
+				path, index, actionKind)
+		}
+
+		// deliver all acks that are ready
+		driver.DeliverAcks()
+
+		if verbose {
+			t.Logf("Current actual state: %s", driver.getStateString())
+		}
+
+		// check that the actual state is the same as the model state
+		t.Logf("Comparing model state to actual state...")
+
+		// compare the running consumers
+		modelRunningConsumers := RunningConsumers(currentModelState)
+
+		systemRunningConsumers := driver.runningConsumers()
+		actualRunningConsumers := make([]string, len(systemRunningConsumers))
+		for i, chain := range systemRunningConsumers {
+			actualRunningConsumers[i] = chain.ChainId
+		}
+
+		// sort the slices so that we can compare them
+		sort.Slice(modelRunningConsumers, func(i, j int) bool {
+			return modelRunningConsumers[i] < modelRunningConsumers[j]
+		})
+		sort.Slice(actualRunningConsumers, func(i, j int) bool {
+			return actualRunningConsumers[i] < actualRunningConsumers[j]
+		})
+
+		require.Equal(t, modelRunningConsumers, actualRunningConsumers, "Running consumers do not match")
+
+		// check validator sets - provider current validator set should be the one from the staking keeper
+		CompareValidatorSets(t, driver, currentModelState, actualRunningConsumers, index)
+
+		// check times - sanity check that the block times match the ones from the model
+		CompareTimes(driver, actualRunningConsumers, currentModelState, timeOffset)
+
+		// check sent packets: we check that the package queues in the model and the system have the same length.
+		for _, consumer := range actualRunningConsumers {
+			ComparePacketQueues(t, driver, currentModelState, consumer, timeOffset)
+		}
+		// compare that the sent packets on the proider match the model
+		CompareSentPacketsOnProvider(driver, currentModelState, timeOffset)
+
+		stats.EnterStats(driver)
+
+		t.Logf("State %v of trace %v is ok!", index, path)
+	}
+	t.Log("🟢 Trace is ok!")
+}
+
+func CompareValidatorSets(t *testing.T, driver *Driver, currentModelState map[string]itf.Expr, consumers []string, index int) {
+	t.Helper()
+	modelValSet := ValidatorSet(currentModelState, "provider")
+	curValSet := driver.providerValidatorSet()
```

Comment:
Maybe just have:
```
actualValSet := make(map[string]int64, len(driver.providerValidatorSet()))
```
`curValSet` might be confused to be related with `currentModelState` due to `cur`?